homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -137,6 +144,35 @@ class SynologyDSMFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
         return self.async_create_entry(title=host, data=config_data,)
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered synology_dsm."""
+        parsed_url = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION])
+        friendly_name = (
+            discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME].split("(", 1)[0].strip()
+        )
+
+        if self._host_already_configured(parsed_url.hostname):
+            return self.async_abort(reason="already_configured")
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["title_placeholders"] = {
+            CONF_NAME: friendly_name,
+            CONF_HOST: parsed_url.hostname,
+        }
+
+        self.discovery_schema = _schema_with_defaults(
+            {CONF_HOST: parsed_url.hostname, CONF_NAME: friendly_name}
+        )
```

Comment:
For a discovery schema, we shouldn't allow changing host/port/ssl. Right now no errors happen because we don't set a unique ID on the config flow. But if we would, then allowing to change the host could lead to issues.

Reply:
Will lock the host.  The port that discovery gives us isn't always the port they actually need to input since it can be changed.

Reply:
No longer asking for the host and name.  May remove name based on the other discussion in this pr.